File: algorithms/c/shuffleArray.c
int* shuffle(int* nums, int numsSize, int n, int* returnSize){
    *returnSize=numsSize;
    int *shuffle=malloc(numsSize*sizeof(int));

    for(int i=0,k=0; i<n; i++,k+=2){
        *(shuffle+k)=nums[i];
        *(shuffle+k+1)=nums[i+(n)];
    }

    return shuffle;
}

File: algorithms/java/shuffleArray.java
class Solution {
    public int[] shuffle(int[] nums, int n) {
        int shuffled[] = new int[nums.length];
        for(int i=0,j=n,k=0; j<nums.length; i++,j++,k+=2){
            shuffled[k] = nums[i];
            shuffled[k+1] = nums[j];
        }
    return shuffled;
    }
}
